BlueDBOnDisk/src/main/java/org/bluedb/disk/collection/BlueCollectionOnDisk.java

```diff
@@ -58,17 +59,24 @@
 	private final IndexManager<T> indexManager;
 
 	public BlueCollectionOnDisk(BlueDbOnDisk db, String name, Class<? extends BlueKey> requestedKeyType, Class<T> valueType, List<Class<? extends Serializable>> additionalRegisteredClasses) throws BlueDbException {
+		this(db, name, requestedKeyType, valueType, additionalRegisteredClasses, null);
+	}
+
+	public BlueCollectionOnDisk(BlueDbOnDisk db, String name, Class<? extends BlueKey> requestedKeyType, Class<T> valueType, List<Class<? extends Serializable>> additionalRegisteredClasses, Long segmentSize) throws BlueDbException {
 		this.valueType = valueType;
 		collectionPath = Paths.get(db.getPath().toString(), name);
+		boolean isNewCollection = !collectionPath.toFile().exists();
 		collectionPath.toFile().mkdirs();
 		metaData = new CollectionMetaData(collectionPath);
 		Class<? extends Serializable>[] classesToRegister = metaData.getAndAddToSerializedClassList(valueType, additionalRegisteredClasses);
 		serializer = new ThreadLocalFstSerializer(classesToRegister);
 		fileManager = new FileManager(serializer);
-		this.keyType = determineKeyType(metaData, requestedKeyType);
+		segmentSize = determineSegmentSize(metaData, requestedKeyType, segmentSize, isNewCollection);
+		keyType = determineKeyType(metaData, requestedKeyType);
+		SegmentSizeSettings segmentSizeSettings = SegmentSizeSettings.getSettings(keyType, segmentSize);
 		recoveryManager = new RecoveryManager<T>(this, fileManager, serializer);
 		rollupScheduler = new RollupScheduler(this);
-		segmentManager = new SegmentManager<T>(collectionPath, fileManager, this, this.keyType);
+		segmentManager = new SegmentManager<T>(collectionPath, fileManager, this, segmentSizeSettings);
 		indexManager = new IndexManager<>(this, collectionPath);

```

Comment:
Should we support being able to specify what segment size settings to use for an index. I don't know that it needs to be configurable by the user, but we may want to set it in the index's meta data and have the ability to change the default segment size settings for different types of indices.

Reply:
probably, but I was planning to do that on a later pull request